### PR TITLE
Add function step tests for function defined outside python environment

### LINKED ISSUE #1556: Create test for function step that executes command in an other python environment
Given a module and function "foo:bar" we need a test that asserts that a `FunctionStep` and the "bar" function can be executed in a python environment without the "foo" module.

This way we can develop and refactor the `FunctionStep` without the fear of breaking this hard-to-test scenario.

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Resolves #1556

### CODE REVIEW

**Review by jondequinor — COMMENTED** [review]
This is cool. I find it a bit odd that you didn't instantiate a `PrefectEnsemble`. Care add some context for that decision?

You can get the resulting output transmitter from the monitor using `get_result` (in the case that was the reason).

**Comment by DanSava** [comment]
> This is cool. I find it a bit odd that you didn't instantiate a `PrefectEnsemble`. Care add some context for that decision?
> 
> You can get the resulting output transmitter from the monitor using `get_result` (in the case that was the reason).

I could have instantiated a `PrefectEnsemble` that is very true, but that would have required a full prefect ensemble config, and I was under the impression that its days are numbered and was thinking of minimizing the changes needed to update the tests when that faithful day actually comes. 

But if you think that using an instance of PrefectEnsemble in the tests would add something extra that the current implementation is missing I am glad to add another test or update the current one.

File: tests/ensemble_evaluator/test_prefect_ensemble.py [hunk 1]
```diff
@@ -456,6 +575,74 @@ def test_run_prefect_ensemble(unused_tcp_port, coefficients):
...
+        transmitted_record = lambda record: asyncio.get_event_loop().run_until_complete(
+            record.load()
+        )
+        expected_results = [
+            pickle.loads(pickle_func)(coeffs) for coeffs in coefficients
+        ]
+        results = [
+            transmitted_record(res["function_output"]).data for res in results.values()
+        ]
+        assert expected_results == results
```
Comment (jondequinor) [inline tests/ensemble_evaluator/test_prefect_ensemble.py:643]: Suggesting a approach slightly more consistent with other `Record.load` consumer code.

```suggestion
        transmitter_futures = [
            res["function_output"].load() for res in results.values()
        ]
        results = asyncio.get_event_loop().run_until_complete(
            asyncio.gather(*transmitter_futures)
        )
        expected_results = [
            pickle.loads(pickle_func)(coeffs) for coeffs in coefficients
        ]
        assert expected_results == [res.data for res in results]
```

**Review by jondequinor — APPROVED** [review]
I like approach #2 better. It is an easier read imo, and it's closer to what I think we should test. Which is that the function step does what it advertises.

Other than that, good job! I like it

**Pull request merged**